### PR TITLE
LOK-2577 Define UEI  for metric threshold

### PR DESCRIPTION
**Pull request by jahanzebdataq**
## Description
Define UEI(e.g. lowThresholdExceeded,highThresholdExceeded,highThresholdRearmed,lowThresholdRearmed) for metric threshold while creating monitoring policy.

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2577

### CODE REVIEW

**Comment by jahanzebdataq** [comment]
Screen shots
Screenshot from 2024-06-12 16-33-57
Screenshot from 2024-06-12 16-33-37
Screenshot from 2024-06-12 16-33-11
Screenshot from 2024-06-12 16-32-53
Screenshot from 2024-06-12 16-32-10
Screenshot from 2024-06-12 16-31-56
Screenshot from 2024-06-12 16-31-38

File: alert/src/main/java/org/opennms/horizon/alertservice/service/MonitorPolicyService.java [hunk 1]
```diff
@@ -148,6 +152,22 @@ public MonitorPolicyProto createPolicy(MonitorPolicyProto request, String tenant
...
+    }
+
+    private void createOrUpdateThresholdMetric(AlertCondition alertCondition) {
+        if (alertCondition.getRule().getEventType().equals(EventType.METRIC_THRESHOLD)
+                && alertCondition.getThresholdMetric() != null) {
+            alertCondition.setThresholdMetric(thresholdMetricRepository.save(alertCondition.getThresholdMetric()));
+            EventDefinition eventDefinition = new EventDefinition();
+            eventDefinition.setEventUei("uei.opennms.org/threshold/highThresholdExceeded");
+            eventDefinition.setEventType(alertCondition.getRule().getEventType());
+            eventDefinition.setId(1140L);
```
Comment (fooker) [inline alert/src/main/java/org/opennms/horizon/alertservice/service/MonitorPolicyService.java:166]: This hard-coded value seems suspicious.

**Review by fooker — CHANGES_REQUESTED** [review]

File: alert/src/main/java/org/opennms/horizon/alertservice/service/MonitorPolicyService.java (hunk 1 above)
Comment (jahanzebdataq) [reply alert/src/main/java/org/opennms/horizon/alertservice/service/MonitorPolicyService.java:166]: As per discussion with @cgorantla for now we are using hard-coded entry for trigger event

**Review by cgorantla — COMMENTED** [review]
Possible to add this alert conditions in DB itself ?

**Comment by jahanzebdataq** [comment]
> Possible to add this alert conditions in DB itself ?

Added

**Comment by jahanzebdataq** [comment]
Merged these changes in LOK-2540

**Pull request closed without merging**